checks/check_extra7150

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+# Prowler - the handy cloud security tool (copyright 2018) by Toni de la Fuente
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy
+# of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+CHECK_ID_extra7150="7.150"
+CHECK_TITLE_extra7150="[extra7150] Check if Elastic Load Balancers have deletion protection enabled"
+CHECK_SCORED_extra7150="NOT_SCORED"
+CHECK_TYPE_extra7150="EXTRA"
+CHECK_SEVERITY_extra7150="Medium"
+CHECK_ASFF_RESOURCE_TYPE_extra7150="AwsElbLoadBalancer"
+CHECK_ALTERNATE_check7150="extra7150"
+CHECK_SERVICENAME_extra7150="elb"
+CHECK_RISK_extra7150='If deletion protection is not enabled, the resource is not protected against deletion.'
+CHECK_REMEDIATION_extra7150='Enable deletion protection attribute; this is not enabled by default.'
+CHECK_DOC_extra7150='https://docs.aws.amazon.com/elasticloadbalancing/latest/application/application-load-balancers.html#deletion-protection'
+CHECK_CAF_EPIC_extra7150='Data Protection'
+
+extra7150(){
+  # "Check if Elastic Load Balancers have delete protection enabled."
+  for regx in $REGIONS; do
+    LIST_OF_ELBSV2=$($AWSCLI elbv2 describe-load-balancers $PROFILE_OPT --region $regx --query 'LoadBalancers[*].LoadBalancerArn' --output text|xargs -n1)
+    if [[ $LIST_OF_ELBSV2 ]]; then
+      for elbarn in $LIST_OF_ELBSV2; do
+        CHECK_DELETION_PROTECTION_ENABLED=$($AWSCLI elbv2 describe-load-balancer-attributes $PROFILE_OPT --region $regx --load-balancer-arn $elbarn --query Attributes[*] --output text|grep -B 1 "deletion_protection.enabled" | grep true)
+        ELBV2_NAME=$(echo $elbarn|cut -d\/ -f3)
```

Comment:
Doing this inside the for-loop might take a lot of time, do we have a workaround to do a for-loop but with the data already obtained?

Reply:
Did you mean for the data already obtained via `describe-load-balancers`? The attributes do not show there unfortunately. For line 32, since there are multiple versions of ELBv2 (NLB, ALB and Gateway LB) the response/output varies depending on which one it is. The [response/output](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/elbv2/describe-load-balancer-attributes.html) is a list of attributes but the order of those varies in the response depending on the type of ELB and therefore I can't extract the boolean directly. That is why I decided to use grep there to determine the boolean expression. The line 33 is just to make it easier to read for the user instead of returning the full Arn but also because the `describe-load-balancer-attributes` needs the full arn and can't use only the name of the ELB to run.

Reply:
For reference, it took a second or so to show data from my 3 deployed ELBs in us-east-1. Please let me know if you have an idea of a workaround I could use.

Reply:
Also, I did notice I can remove` -B 1` from `grep -B 1 "deletion_protection.enabled"` so I will make that update too before the next update. I didnt say it before but thank you for the feedback!